accepted/00075-live-patching-enhancements.md

```diff
@@ -0,0 +1,170 @@
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+
+### Data Retrieval
+
+This section evaluates three alternative implementations for this feature.
+
+#### Call 'zypper lifecycle' on client
+
+Uyuni calls `zypper lifecycle` on the client and collects the data via Salt during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - `zypper lifecycle` is publicly supported
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - `zypper lifecycle` currently provides no machine-readable format
+ - Extra packages need to be installed on the client:
+   - `zypper-lifecycle-plugin`
+   - `lifecycle-data-sle-module-live-patching`
+
+#### Read the lifecycle data file on client
+
+Uyuni runs a Salt state to read the lifecycle data file provided with the package `lifecycle-data-sle-module-live-patching` on the client and collects the data during package profile update. The data is then stored in the database per client along with the kernel live version data.
+
+**Pros:**
+ - Easy to implement
+ - Can be attached to profile update data with an existing process
+
+**Cons:**
+ - File name or format can change in the future
+ - `lifecycle-data-sle-module-live-patching` package must be installed on the client
+
+#### Unpack and read the lifecycle data RPM on Uyuni server
+
+The lifecycle information is provided in the `lifecycle-data-sle-module-live-patching` package in the Live Patching product. Uyuni unpacks this RPM and reads the data from the extracted CSV file during reposync of this product. The data is stored in the database per live patch package.
+
+**Pros:**
+ - Independent, no impositions on clients
+ - Gathered data is always available and can be used for UI enhancements such as displaying EOL info for all live patch packages displayed in the UI
+
+**Cons:**
+ - RPM format can change in the future
+ - Uyuni doesn't have any existing functionality that can be utilized
+ - Unpacking must be done during reposync, adding complextion
+
+## 2. Live patching CLM workflow
+
+SLES channels can be prepared for live patching applications using the Content Lifecycle Management feature in Uyuni. This is done by adding a set of filters that exclude the regular kernel package updates and include live patch versions of those. However, the UI doesn't provide any special workflow for live patching and this process must be done manually by the user.
+
+Below are some examples of such sets of filters:
+
+Live patching project with a single filter to exclude regular kernel packages newer than a specific version:
+ - `DENY erratum contains package with epoch/version/release greater than kernel-default <EVR> (package_nevr)`
+
+Live patching project using `ALLOW` overrides:
+ - `DENY errata containing "update for the Linux Kernel" in synopsis`
```

Comment:
I think this was a DENY of all patches which suggest a reboot.

Reply:
IIRC, Don demoed two different setups, one like this, and one using the newer reboot filter.

Reply:
Btw: Example B is potentionally dangerous as we create 2 Filters which depend on the Patch Synopsis. This is a free text field and we cannot rely on Maintenance Team always stick to the same naming conventions. Especially as this synopsis is autogenerated from the buildservice and manually updated by them. When they forget it of do a typo our filter might not match anymore.

Reply:
Did I get it correctly that with the new metadata we should be able to (optionally automatically) create filters that do not depend on relatively-fragile synopsis like the one described here?

Reply:
No. The new metadata are for knowing when there will be no live-patches anymore for the installed kernel and it is time to switch to the new kernel.
These filters are for general use of life-patching feature. Stuff we already have.
But we want to add a new UI to make it easier to get the filters. But for this we need to know which filters we want to set. There are multiple ways to achive the wanted result. We described Example A in the manuals, but Don took a different approach (Example B).

What I want to say is: it does not make sense if we implement a UI for Example A but Customers like more Example B and do not use our new shiny UI. 
So we need to find out why people like Example B and if we can convince them to use Example A or modify it and make an Example C which please them all :-)

Reply:
I'll start an email thread to discuss all the different options with supporters